hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -73,7 +72,7 @@ object PruneDeadFields {
           rebuild(bmir, memo)
         case vir: IR =>
           memoizeValueIR(vir, vir.typ, memo)
-          rebuild(vir, Env.empty[Type], memo)
+          rebuildIR(vir, BindingEnv(Env.empty, Some(Env.empty), Some(Env.empty)), memo)
```

Comment:
Why are both of these some?  If this is the initial entrypoint, I'd expect both of these to be None, no?

Reply:
All of our AggregatorSuite tests break without this. We optimize naked value IRs with aggregations all the time.

Reply:
I looked at what it would take to thread the allowed contexts through the optimizer and interpreter and decided that it wasn't worth it

Reply:
The rebuilding Env is the Env of Ref replacements. Since the types that things get coming up on the rebuild pass are subject to `minimal type < T < full type`, we need an env of the `T` types